### PR TITLE
Fix: dedicated Discogs hander

### PR DESCRIPTION
**Pull request by felipap**
Discogs handler is broken, as reported by #43.

Creates a dedicated handler for Discogs.

There are several other handlers currently stuck on OAuth 1.0 which we'll have to fix.

### CODE REVIEW

File: handshake/src/providers/discogs.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+ * ```
+ *
+ * @providersetup
+ *
+ * @troubleshoot
+ */
+export const Discogs: HandlerFactory<Args> = (args) => {
+  assert(args.consumerKey, "consumerKey is empty or missing");
+  assert(args.consumerSecret, "consumerSecret is empty or missing");
+
```
Comment (kilinkis) [inline handshake/src/providers/discogs.ts:37]: Useful assertions here! they already helped me when testing this branch.

File: handshake/src/providers/discogs.ts [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+      method: "POST",
+      headers: {
+        "User-Agent": "handshake/1.0",
+        Authorization: "OAuth",
+      },
+      body: new URLSearchParams({
+        oauth_nonce: nonce,
+        oauth_token: oauthToken,
+        oauth_consumer_key: consumerKey,
+        oauth_signature: consumerSecret + "&" + tokenSecret,
```
Comment (kilinkis) [inline handshake/src/providers/discogs.ts:228]: good find! I think it is not documented (by Discogs) that this field uses these two values

**Review by kilinkis — APPROVED** [review]
Works! thank you so much for taking the time to fixing this 🗽

image

**Comment by felipap** [comment]
Thanks for the review @kilinkis! Merging this. 👍

**Pull request merged**